src/features/settings/rpc/setOfficialRole.tsx

```diff
@@ -29,8 +29,9 @@ async function handle(params: Params, apiClient: IApiClient): Promise<Result> {
     { role: role }
   );
 
-  const membership = await apiClient.get<[ZetkinMembership]>(
+  const memberships = await apiClient.get<[ZetkinMembership]>(
     `/api/orgs/${orgId}/people/${official.id}/connections`
   );
-  return membership[0];
+
+  return memberships.find((m) => m.organization.id == orgId)!;
```

Comment:
I'm not loving the non-nullish assertion operator here. It is _extremely_ unlikely that this will ever be anything other than a membership, but in theory, it could be. So I think it's better to handle the null case. Not here though, because you have nothing else to return.

So I think the `Result` type should be changed to allow `null` and then you have to handle that case on the front-end, (e.g. by doing nothing, because this will be so unlikely).